From 67465fb930db2c9194c9c7ac81688217380eaf16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Fri, 26 Oct 2012 15:10:29 +0300
Subject: [PATCH 09/10] Do not build webkit2

Remove WebKit2 (split-process plugin architecture) from the build
entirely. This change seems to be enough to fit the library linking
within 32-bit address space.
---
 Source/QtWebKit.pro                          |    9 ---------
 Source/WebKit/qt/declarative/declarative.pro |    6 ------
 Source/api.pri                               |    9 ---------
 Tools/Tools.pro                              |    8 --------
 WebKit.pro                                   |    6 ------
 5 files changed, 38 deletions(-)

diff --git a/Source/QtWebKit.pro b/Source/QtWebKit.pro
index d71cdda..6379900 100644
--- a/Source/QtWebKit.pro
+++ b/Source/QtWebKit.pro
@@ -10,15 +10,6 @@ CONFIG += ordered
 api.file = api.pri
 SUBDIRS += api
 
-!no_webkit2 {
-    webprocess.file = WebKit2/WebProcess.pro
-    SUBDIRS += webprocess
-    contains(DEFINES, ENABLE_PLUGIN_PROCESS=1) {
-        pluginprocess.file = WebKit2/PluginProcess.pro
-        SUBDIRS += pluginprocess
-    }
-}
-
 include(WebKit/qt/docs/docs.pri)
 
 declarative.file = WebKit/qt/declarative/declarative.pro
diff --git a/Source/WebKit/qt/declarative/declarative.pro b/Source/WebKit/qt/declarative/declarative.pro
index 04f2298..7b0753d 100644
--- a/Source/WebKit/qt/declarative/declarative.pro
+++ b/Source/WebKit/qt/declarative/declarative.pro
@@ -10,9 +10,3 @@ CONFIG += ordered
 public_api.file = public.pri
 public_api.makefile = Makefile.declarative.public
 SUBDIRS += public_api
-
-!no_webkit2 {
-    experimental_api.file = experimental/experimental.pri
-    experimental_api.makefile = Makefile.declarative.experimental
-    SUBDIRS += experimental_api
-}
diff --git a/Source/api.pri b/Source/api.pri
index 4ac64c3..9e5a500 100644
--- a/Source/api.pri
+++ b/Source/api.pri
@@ -74,15 +74,6 @@ WEBKIT += javascriptcore
 
 WEBKIT += webcore
 
-!no_webkit2 {
-    WEBKIT += webkit2
-
-    # Ensure that changes to the WebKit1 and WebKit2 API will trigger a qmake of this
-    # file, which in turn runs syncqt to update the forwarding headers.
-    QMAKE_INTERNAL_INCLUDED_FILES *= WebKit2/Target.pri
-    QMAKE_INTERNAL_INCLUDED_FILES *= WebKit/WebKit1.pro
-}
-
 !no_webkit1: WEBKIT += webkit1
 
 qnx {
diff --git a/Tools/Tools.pro b/Tools/Tools.pro
index 749013c..31bfad2 100644
--- a/Tools/Tools.pro
+++ b/Tools/Tools.pro
@@ -13,14 +13,6 @@ CONFIG += ordered
     SUBDIRS += DumpRenderTree/qt/ImageDiff.pro
 }
 
-!no_webkit2 {
-    # WTR's InjectedBundle depends currently on WK1's DumpRenderTreeSupport
-    !no_webkit1:contains(DEFINES, HAVE_QTQUICK=1):contains(DEFINES, HAVE_QTTESTLIB=1): SUBDIRS += WebKitTestRunner/WebKitTestRunner.pro
-
-    contains(DEFINES, HAVE_QTQUICK=1): SUBDIRS += MiniBrowser/qt/MiniBrowser.pro
-    SUBDIRS += MiniBrowser/qt/raw/MiniBrowserRaw.pro
-}
-
 !win32:contains(DEFINES, ENABLE_NETSCAPE_PLUGIN_API=1) {
     SUBDIRS += DumpRenderTree/qt/TestNetscapePlugin/TestNetscapePlugin.pro
 }
diff --git a/WebKit.pro b/WebKit.pro
index 4ace815..320e8a2 100644
--- a/WebKit.pro
+++ b/WebKit.pro
@@ -68,12 +68,6 @@ SUBDIRS += WebCore
     SUBDIRS += webkit1
 }
 
-!no_webkit2 {
-    webkit2.file = Source/WebKit2/WebKit2.pro
-    webkit2.makefile = Makefile.WebKit2
-    SUBDIRS += webkit2
-}
-
 QtWebKit.file = Source/QtWebKit.pro
 QtWebKit.makefile = Makefile.QtWebKit
 SUBDIRS += QtWebKit
-- 
1.7.10.4

